dace/codegen/instrumentation/data/data_dump.py

```diff
@@ -106,6 +106,11 @@ def on_node_end(self, sdfg: SDFG, state: SDFGState, node: nodes.AccessNode, oute
             # Only run on host code
             return
 
+        condition_preamble, condition_postamble = '', ''
+        if node.instrument_condition is not None and not node.instrument_condition.as_string == '1':
+            condition_preamble = f'if ({node.instrument_condition.as_string})' + ' {'
```

Comment:
Please check how we generate C++ from codeblocks

Reply:
Learned something new, thanks.. :-)